### PR TITLE
Cw 150 cake pay introduction card

### PR DESCRIPTION
**Pull request by Serhii-Borodenko**
add an introduction card about Cake Pay service, for XMR, BTC, LTC wallets

### CODE REVIEW

File: lib/view_model/dashboard/balance_view_model.dart [hunk 1]
```diff
@@ -235,6 +237,12 @@ abstract class BalanceViewModelBase with Store {
   @computed
   CryptoCurrency get currency => appStore.wallet.currency;
 
+  @observable
+  bool isShowCard = true;
```
Comment (mkyq) [inline lib/view_model/dashboard/balance_view_model.dart:241]: Please value in constructor

File: lib/view_model/dashboard/balance_view_model.dart [hunk 2]
```diff
@@ -246,6 +254,16 @@ abstract class BalanceViewModelBase with Store {
     _onCurrentWalletChangeReaction?.reaction?.dispose();
   }
 
+  @action
+  Future<void> disableIntroCakePayCard () async {
+    final value = wallet.walletInfo;
+    final  box = Hive.box<WalletInfo>('WalletInfo');
```
Comment (mkyq) [inline lib/view_model/dashboard/balance_view_model.dart:260]: Inject hive box with DI

File: lib/view_model/wallet_creation_vm.dart [hunk 3]
```diff
@@ -37,6 +37,9 @@ abstract class WalletCreationVMBase with Store {
   bool nameExists(String name)
     => walletCreationService.exists(name);
 
+  bool typeExists(WalletType type)
+  => walletCreationService.typeExists(type);
```
Comment (mkyq) [inline lib/view_model/wallet_creation_vm.dart:41]: Add space to arrow function sybmols as it's in function above

File: lib/view_model/dashboard/balance_view_model.dart [hunk 4]
```diff
@@ -246,6 +254,16 @@ abstract class BalanceViewModelBase with Store {
...
+    final  currentWallet = box.get(value.key);
+    currentWallet.showIntroCakePayCard = false;
+    await box.put(value.key, currentWallet);
```
Comment (mkyq) [inline lib/view_model/dashboard/balance_view_model.dart:263]: Actually here you can use `await save()` on walletInfo for update information

**Review by mkyq — CHANGES_REQUESTED** [review]

**Comment by Serhii-Borodenko** [comment]
done

**Comment by Serhii-Borodenko** [comment]
@mkyq

**Review by mkyq — APPROVED** [review]

**Pull request merged**